### ISSUE TITLE
Docker-squash not properly working

### ISSUE THREAD

**Issue by samprog1** [jwilder/docker-squash#65]
With the command `docker save d5b63701456a | sudo /usr/local/bin/docker-squash -verbose -t test:squashed | docker load` I don't get any new images, though there's no error.

Output:

```
Loading export from STDIN using /tmp/docker-squash299353940 for tempdir
Loaded image w/ 13 layers
Extracting layers...
  -  /tmp/docker-squash299353940/f3376dbeb3b0e4f36439bca922ec3cd370ef05c5243a37f038170716d556aeb3/layer.tar
  -  /tmp/docker-squash299353940/09967547b5df888f5e066a52d72526b1c68f6837e216799c91c26499dd42c6a7/layer.tar
  -  /tmp/docker-squash299353940/457485a3a84745e213bbb7fc065eacc2826c1d0dacc8743dcb47383b9140b248/layer.tar
  -  /tmp/docker-squash299353940/a238fce9a846a5209b8c1d0bc4c620b4cac48fb512c5b391f4bc655a80b3aa79/layer.tar
  -  /tmp/docker-squash299353940/a4d961cdf6bb153d27f8f58f8e2eaec75241ac60e51dd39aecbc34f2164db202/layer.tar
  -  /tmp/docker-squash299353940/ac03b9ff3e20837ea0e0095e093ee76c8bcffc18fb29ce5d70cc317c67129cef/layer.tar
  -  /tmp/docker-squash299353940/b13b4991fd92f5c3f893ec70993dca32bdc80785030e538b575fa68612a9c4e9/layer.tar
  -  /tmp/docker-squash299353940/bf6a503f7a3e7527729622ed72da1999d3d8b5a674ba3e18cd4ef1e01d1b5c97/layer.tar
  -  /tmp/docker-squash299353940/f30d7a68d1426e11ddea1ea508144e78cd1581f2aa710a28bc1157da72d659bb/layer.tar
  -  /tmp/docker-squash299353940/15ca8d025fb12dd0390b50364bf71b5573ba07a4d92d266cf23b7c3c29a87e4b/layer.tar
  -  /tmp/docker-squash299353940/17215e1d258f4f6306eec1764f142fc4fd3cf85d050a88b1580e16ae98626cb6/layer.tar
  -  /tmp/docker-squash299353940/3b5667d6dbc444eb8baaac05ca46bdc18f6d872d0c8d899b9064749a1cbe80f0/layer.tar
  -  /tmp/docker-squash299353940/626f3bd3ed630a7876e2f1f002b7ea8a4020cacf3e0319498831ee2c604c6ee7/layer.tar
Inserted new layer 4455bd2b5c88 after 15ca8d025fb1
  -  15ca8d025fb1
  -> 4455bd2b5c88 /bin/sh -c #(squash) from 15ca8d025fb1
  -  3b5667d6dbc4
  -  626f3bd3ed63
  -  17215e1d258f
  -  ac03b9ff3e20
  -  bf6a503f7a3e
  -  457485a3a847
  -  f3376dbeb3b0
  -  a238fce9a846
  -  b13b4991fd92
  -  a4d961cdf6bb
  -  09967547b5df
  -  f30d7a68d142 /bin/sh -c #(nop) CMD ["/bin/sh" "-c" "node /srv/node/Centra
Squashing from 4455bd2b5c88 into 4455bd2b5c88
  -  Deleting whiteouts for layer 3b5667d6dbc4
  -  Deleting whiteouts for layer 626f3bd3ed63
  -  Deleting whiteouts for layer 17215e1d258f
  -  Deleting whiteouts for layer ac03b9ff3e20
  -  Deleting whiteouts for layer bf6a503f7a3e
  -  Deleting whiteouts for layer 457485a3a847
  -  Deleting whiteouts for layer f3376dbeb3b0
  -  Deleting whiteouts for layer a238fce9a846
  -  Deleting whiteouts for layer b13b4991fd92
  -  Deleting whiteouts for layer a4d961cdf6bb
  -  Deleting whiteouts for layer 09967547b5df
  -  Deleting whiteouts for layer f30d7a68d142
  -  Rewriting child history
  -  Removing 3b5667d6dbc4. Squashed. ()
  -  Removing 626f3bd3ed63. Squashed. ()
  -  Removing 17215e1d258f. Squashed. ()
  -  Removing ac03b9ff3e20. Squashed. ()
  -  Removing bf6a503f7a3e. Squashed. ()
  -  Removing 457485a3a847. Squashed. ()
  -  Removing f3376dbeb3b0. Squashed. ()
  -  Removing a238fce9a846. Squashed. ()
  -  Removing b13b4991fd92. Squashed. ()
  -  Removing a4d961cdf6bb. Squashed. ()
  -  Removing 09967547b5df. Squashed. ()
  -  Replacing f30d7a68d142 w/ new layer 9d2df5ab1cca (/bin/sh -c #(nop) CMD ["/bin/sh" "-c" "node /sr...)
Tarring up squashed layer 4455bd2b5c88
Removing extracted layers
Tagging 9d2df5ab1cca as test:squashed
Tarring new image to STDOUT
Done. New image created.
  -  15ca8d025fb1 292.471209 years  130.9 MB
  -  4455bd2b5c88 26 seconds /bin/sh -c #(squash) from 15ca8d025fb1 640.8 MB
  -  9d2df5ab1cca 16 seconds /bin/sh -c #(nop) CMD ["/bin/sh" "-c" "node /srv/node/Cen... 70.64 MB
Removing tempdir /tmp/docker-squash299353940
```

When docker-squashing into an output file and trying to load it from that, I get the following error:
`$ docker load -i squashed.tar
Error response from daemon: open /var/lib/docker/tmp/docker-import-102261036/7dc29a8f47df2223a335cd818b446df8cdde86863a46c13a18211916eb5a759c/layer.tar: no such file or directory`

I'm using SLES 12 SP 1 Docker version 1.10.3

Any ideas?

**Comment by kpeu3i** [comment]
+1

I have the same issue

**Comment by andrea-spoldi** [comment]
+1

**Comment by miggy282** [comment]
+1 same issue  docker images doesn't  show new squashed image

**Comment by baristahell** [comment]
+1 here, not getting anything after shashing. 

``
docker save IM_ID | sudo docker-squash -from root -t me/repo:tag | docker load
`` 
``
Loaded image ID: sha256:IM_IDnumbers
``

**Comment by hakatashi** [comment]
Same issue (docker images doesn't show new squashed image), and the command given in https://github.com/jwilder/docker-squash/issues/53#issuecomment-222045035 worked fine for me.

    tar --delete -f squashed.tar manifest.json

**Comment by ChrisGClark** [comment]
same issue for me (running on mac os).   

I do the piped commands:    save..  | sudo  docker-squash... | docker load 
works without errors

but 'docker images'   does not show new image.   
If it type docker images <id from end of docker load command, after sha256: >  no images

**Comment by ppridge** [comment]
Same issue for me.  There are no errors in the squashing process.   After completion  docker images does not show any new images.   I have tried step by step (save, squash, load) and piped (save | squach | load) same result.   I have tried several different images and have not yet been able to see a squashed image in docker image repo.   Here is my docker version:

Docker version 1.12.1, build 23cf638
[

**Comment by opichon** [comment]
Same issue for me, whether using the piped commands or the individual commands separately.

**Comment by scytherswings** [comment]
Has this project been abandoned? I am also having the same issue. What a shame.

**Comment by mauron85** [comment]
Having same problem. Not an expert, but it seems docker load is importing actually saved imaged not squashed.

```
docker save ab4e6a73e308 | sudo TMPDIR=/home/finch/tmp docker-squash -t ubuntu-xenial-build-essential:squash -verbose -from root | docker load
```

Output
```
Tagging 5adf140b9b41 as ubuntu-xenial-build-essential:squash
Tarring new image to STDOUT
Done. New image created.
  -  561dcc9b249f 47.499343 years  132.8 MB
  -  5adf140b9b41 3 seconds /bin/sh -c #(squash) from 561dcc9b249f 303.4 MB

Loaded image ID: sha256:ab4e6a73e3081be5d0decd163ab4034bdb33e5b5a7c72d35f8ec9d5bb2b93476
```

As you can see sha of loaded image is same as exported image (`docker save ab4e6a73e308 ...`), but IMHO it should be `561dcc9b249f`.

Could it be that unix pipe is broken somewhere, so instead of result of docker-squash | docker-load, docker save is actually piped (docker save ab4e6a73e308 | broken pipe | docker load ?

**Comment by jwilder** [comment]
Docker changed how it calculated layer IDs in 1.10: https://blog.docker.com/2016/01/docker-1-10-rc/

`docker-squash` was built using the original version and broken around this time.  I haven't had time to update the project to use the new hashing model.  There was a start to fix this in #55, but it's not complete.

Docker has added experimental squash support as  `docker build --squash`. See [announcement](https://blog.docker.com/2017/01/whats-new-in-docker-1-13/).  Also, the [multi-stage build](https://docs.docker.com/engine/userguide/eng-image/multistage-build/) support recently introduced will also make squashing less necessary.

**Comment by ppridge** [comment]
Thanks for the update.
tom
